delta/block.go

```diff
@@ -0,0 +1,52 @@
+package delta
+
+import (
+	"github.com/pyroscope-io/client/delta/internal/pprof"
+	"io"
+	"runtime"
+	"sort"
+)
+
+type BlockProfiler struct {
+	impl           pprof.DeltaMutexProfiler
+	runtimeProfile func([]runtime.BlockProfileRecord) (int, bool)
+	scaleProfile   func(int64, float64) (int64, float64)
+}
+
+func NewMutexProfiler() BlockProfiler {
+	return BlockProfiler{runtimeProfile: runtime.MutexProfile, scaleProfile: scaleMutexProfile}
+}
+func NewBlockProfiler() BlockProfiler {
+	return BlockProfiler{runtimeProfile: runtime.BlockProfile, scaleProfile: scaleBlockProfile}
+}
+
+func (d *BlockProfiler) Profile(w io.Writer) error {
+	var p []runtime.BlockProfileRecord
+	n, ok := d.runtimeProfile(nil)
+	for {
+		p = make([]runtime.BlockProfileRecord, n+50)
+		n, ok = d.runtimeProfile(p)
+		if ok {
+			p = p[:n]
+			break
+		}
+	}
+
+	//todo is it needed? use generics when available
+	sort.Slice(p, func(i, j int) bool { return p[i].Cycles > p[j].Cycles })
```

Comment:
It might be used in `pprof` tool – is worth checking. But I think it is better to leave it as is just for better compatibility.

I'm afraid generics are not helpful here. But implementing sort interface on `[]runtime.BlockProfileRecord` and use of `sort.Sort` would slightly increase performance.

Reply:
ok, lets keep it, I will remove the comment

Reply:
my comment was about https://pkg.go.dev/golang.org/x/exp/slices#Sort